From 4ffe31e2a407bc50bbe78b1dbb6548e34076b5ae Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Tue, 27 Jul 2021 15:10:17 +0800
Subject: [PATCH] openeuler_defconfig: Enable Phytium FT-2500 support by
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 4ffe31e2a407bc50bbe78b1dbb6548e34076b5ae

 default

hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I41AUQ
CVE: NA

-------------------------------------------------

While compiling with openeuler_defconfig, configs CONFIG_ARCH_PHYTIUM
and CONFIG_ARM_GIC_PHYTIUM_2500 should be set to enable support for
Phytium ARMv8 S2500/FT-2500 SoC and 2-Processor server.

Enable these two configs for arch/arm64/configs/openeuler_defconfig
by default.

Note: As FT-2500 2-Processor server has 16 NUMA nodes, CONFIG_NODES_SHIFT
should be 4.

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/configs/openeuler_defconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/configs/openeuler_defconfig b/arch/arm64/configs/openeuler_defconfig
index 0ff8e6ce6b78..248ceeefec7f 100644
--- a/arch/arm64/configs/openeuler_defconfig
+++ b/arch/arm64/configs/openeuler_defconfig
@@ -299,6 +299,7 @@ CONFIG_ARCH_HISI=y
 # CONFIG_ARCH_MESON is not set
 # CONFIG_ARCH_MVEBU is not set
 # CONFIG_ARCH_MXC is not set
+CONFIG_ARCH_PHYTIUM=y
 CONFIG_ARCH_QCOM=y
 # CONFIG_ARCH_REALTEK is not set
 # CONFIG_ARCH_RENESAS is not set
@@ -5912,6 +5913,7 @@ CONFIG_ARM_GIC_V2M=y
 CONFIG_ARM_GIC_V3=y
 CONFIG_ARM_GIC_V3_ITS=y
 CONFIG_ARM_GIC_V3_ITS_PCI=y
+CONFIG_ARM_GIC_PHYTIUM_2500=y
 # CONFIG_AL_FIC is not set
 CONFIG_HISILICON_IRQ_MBIGEN=y
 CONFIG_PARTITION_PERCPU=y
-- 
2.26.2

